src/Workspaces/CSharp/Portable/CodeGeneration/ExpressionGenerator.cs

```diff
@@ -255,14 +258,19 @@ private static ExpressionSyntax GenerateSingleLiteralExpression(ITypeSymbol type
                 }
             }
 
+            var signedIntegerMinValue = isSignedIntegerMinValue(value);
             var negative = isNegative(value);
-            if (negative)
+
+            if (negative && !signedIntegerMinValue)
             {
                 value = negate(value);
             }
```

Comment:
instead of checking "isSignedIntegerMin" you could do `var negated = negate(value); if (negated.Equals(value)) { /*special logic*/ }` right?

Reply:
Yup, that makes sense. Good catch